File: algorithms/easy/transformed_array.rs
impl Solution {
    pub fn construct_transformed_array(nums: Vec<i32>) -> Vec<i32> {
        let mut res = vec![0; nums.len()];
        let l = nums.len() as i32;
        for i in 0..nums.len() {
            res[i] = nums[(i as i32 + nums[i]).rem_euclid(l) as usize];
        }
        res
    }
}

File: algorithms/easy/transformed_array.kt
class Solution {
  fun ndx(i: Int, offset: Int, size: Int): Int {
    return ((i + offset) % size + size) % size
  }

  fun constructTransformedArray(nums: IntArray): IntArray {
    val ans = IntArray(nums.size)
    for (i in nums.indices) {
      val j = ndx(i, nums[i], nums.size)
      ans[i] = nums[j]
    }
    return ans
  }
}
